dev/examples/ThreeVariant.tsx

```diff
@@ -0,0 +1,58 @@
+import * as React from "react"
+import { MotionConfig, motion as motionDom } from "framer-motion"
+import { motion, MotionCanvas } from "framer-motion-3d"
+import { Canvas, extend } from "@react-three/fiber"
+import {
+    AmbientLight,
+    PointLight,
+    Group,
+    BoxGeometry,
+    MeshStandardMaterial,
+    Mesh,
+} from "three"
+
+extend({
+    AmbientLight,
+    PointLight,
+    Group,
+    BoxGeometry,
+    MeshStandardMaterial,
+    Mesh,
+})
```

Comment:
Why do you need this? All these types should come in from `@react-three/fiber` automatically

Reply:
Since 8 I think they need pulling in manually https://docs.pmnd.rs/react-three-fiber/tutorials/v8-migration-guide#tree-shaking-via-extend

Reply:
I've been using 8 for a while and never had this? I use it quite often. You also don't see it in recent sandboxes by their team https://codesandbox.io/s/meshepoxymaterial-hmgdjq?file=/src/App.js

I think what you linked to is an example if you want to do more aggressive tree shaking? Or maybe all these examples have some babel plugin hidden that you don't have in your setup here?

Reply:
I've never used extend for these and I use framer-motion-3d and r3f with nextjs, vite, etc often, something must be up. Does it break if you remove?

Reply:
Its just for tree shaking right?

Reply:
@huntercaron Yeah I get an error otherwise. It used to point to the page I've linked but now it only mentions third-party components. Maybe I'll reach out to Paul see if he knows what's going on